extension/js/background.js

```diff
@@ -340,8 +330,30 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
             }
             else if (!(request.savedArticles in allSavedArticles)){
                 console.log('NORMAL CASES')
-                details.savedArticlesCodeZero.savedArticles.push(request.savedArticles);
-                chrome.storage.local.set({'savedArticlesCodeZero':details.savedArticlesCodeZero})
+                console.log(request.savedArticles)
+                console.log(allSavedArticles)
+                let len = allSavedArticles.length;
+                let i =0;
+                let j= 0;
+                for(i=0;i<len;i++){
+                    console.log(i)
+                    console.log(request.savedArticles.URL)
+                    console.log(allSavedArticles[i].URL)
+                    if(request.savedArticles.URL === allSavedArticles[i].URL && i == len-1){
+                        j=10;
```

Comment:
j =10; for what??

Reply:
To have a check whether the loop is terminated at i = allSavedArticles.length, it is an extra condition. j=10, the value is specified only for a check, any value can be specified.

Reply:
try using boolean expression for the same. Boolean is made for the purpose.